### ISSUE TITLE
x/pkgsite: better discoverability for extension interfaces

### ISSUE THREAD

**Issue by josharian** [golang/go#44301]
I was reading https://pkg.go.dev/io/fs, trying to find all the extension interfaces available for fs.FS. I had to scan the entire package API. It would have been even harder had the package not used a consistent naming convention.

Now that the standard library is using it, I suspect we'll see increasing uptake of [the extension pattern](https://go.googlesource.com/proposal/+/master/design/draft-iofs.md#extension-interfaces-and-the-extension-pattern). It'd be nice if pkg.go.dev could recognize it and surface extension interfaces, either through different grouping or through an explicit list at the base interface type.

**Comment by BOOGLE360** [comment]
yes yes milestone it is i see! love it. hard work of learning does pay off
